app.py

```diff
@@ -0,0 +1,423 @@
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Flame app to send sequences to review.
+"""
+
+import os
+import sys
+import uuid
+import sgtk
+from sgtk import TankError
+
+from sgtk.platform import Application
+
+class FlameReview(Application):
+    """
+    Review functionality to automate and streamline sequence review out of flame.
+    
+    Generates quicktimes for the selected flame sequences and uploads these to Shotgun.
+    """
+    
+    def init_app(self):
+        """
+        Called as the application is being initialized.
+        """
+        self.log_debug("%s: Initializing" % self)
+
+        # register our desired interaction with flame hooks
+        menu_caption = self.get_setting("menu_name")
+        
+        # track the comments entered by the user
+        self._review_comments = ""
+        
+        # flag to indicate that something was actually submitted
+        self._submission_done = False
+        
+        # set up callbacks for the engine to trigger 
+        # when this profile is being triggered 
+        callbacks = {}
+        callbacks["preCustomExport"] = self.pre_custom_export
+        callbacks["preExportAsset"] = self.adjust_path
+        callbacks["postExportAsset"] = self.register_post_asset_job
+        callbacks["postCustomExport"] = self.display_summary
+        
+        # register with the engine
+        self.engine.register_export_hook(menu_caption, callbacks)
+        
+    def pre_custom_export(self, session_id, info):
+        """
+        Hook called before a custom export begins. The export will be blocked
+        until this function returns. This can be used to fill information that would
+        have normally been extracted from the export window.
+        
+        :param session_id: String which identifies which export session is being referred to.
+                           This parameter makes it possible to distinguish between different 
+                           export sessions running if this is needed (typically only needed for
+                           expert use cases).
+                           
+        :param info: Dictionary with info about the export. Contains the keys
+                     - destinationHost: Host name where the exported files will be written to.
+                     - destinationPath: Export path root.
+                     - presetPath: Path to the preset used for the export.
+                     - abort: Pass True back to flame if you want to abort
+                     - abortMessage: Abort message to feed back to client
+        """
+        from PySide import QtGui, QtCore
+        
+        # clear our flags
+        self._submission_done = False
+        
```

Comment:
Should the info dictionary be cleared before popping up the dialog ? Could it contain "abort" : True from previous aborted exports, and as we don't set it to False below, this could cause the export to be canceled, even if the user wanted it to go ahead ?